llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -80,6 +80,11 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST) {
       .maxScalar(0, sXLen)
       .lower();
 
+  getActionDefinitionsBuilder(G_CTPOP)
+      .maxScalar(0, sXLen)
```

Comment:
There's an assumption here that the maximum bitcount for any possible value including illegal will fit in XLen bits. This should be true since LLVM's maximum integer type is i16777215.

Clamping the result type first gives better code when splitting s64 intoto two s32 ops on RV32. That will generate an s32 G_ADD to add the two ctpops together. Otherwise we would get an s64 G_ADD that also needs to split.

Reply:
It might be helpful to add this information in a comment in the code?